modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/ScanQueryIterator.java

```diff
@@ -0,0 +1,454 @@
+                            key0,
+                            val0,
+                            null,
+                            null));
+                    }
+
+                    if (transform != null) {
+                        try {
+                            next0 = transform.apply(new CacheQueryEntry<>(key0, val0));
+                        }
+                        catch (Throwable e) {
+                            throw new IgniteException(e);
+                        }
+                    }
+                    else
+                        next0 = (R)(!locNode ? new T2<>(key0, val0) :
+                            new CacheQueryEntry<>(key0, val0));
+
+                    break;
+                }
+            }
+        }
+
+        if ((this.next = next0) == null && expiryPlc != null && dht != null) {
+            dht.sendTtlUpdateRequest(expiryPlc);
+
+            expiryPlc = null;
+        }
+    }
+
+    /** */
+    @Nullable public IgniteBiPredicate<K, V> filter() {
+        return scanFilter;
+    }
+
+    /** */
+    public AffinityTopologyVersion topVer() {
+        return topVer;
+    }
+
+    /** */
+    public GridDhtLocalPartition localPartition() {
+        return locPart;
+    }
+
+    /** */
+    public IgniteClosure<Cache.Entry<K, V>, R> transformer() {
+        return transform;
+    }
+
+    /** */
+    public long startTime() {
+        return startTime;
+    }
+
+    /** */
+    public boolean local() {
+        return locNode;
+    }
+
+    /** */
+    public boolean keepBinary() {
+        return keepBinary;
+    }
+
+    /** */
+    public UUID subjectId() {
+        return subjId;
+    }
+
+    /** */
+    public String taskName() {
+        return taskName;
+    }
+
+    /** */
+    public GridCacheContext<K, V> cacheContext() {
+        return cctx;
+    }
+
+    /** */
+    public int pageSize() {
+        return pageSize;
+    }
+
+    /** */
+    private @Nullable IgniteBiPredicate<K, V> prepareScanFilter(IgniteBiPredicate<K, V> keyValFilter) throws IgniteCheckedException {
+        if (keyValFilter == null)
+            return null;
+
+        try {
+            if (keyValFilter instanceof PlatformCacheEntryFilter)
+                ((PlatformCacheEntryFilter)keyValFilter).cacheContext(cctx);
+            else
+                injectResources(keyValFilter, cctx);
+
+            return SecurityUtils.sandboxedProxy(cctx.kernalContext(), IgniteBiPredicate.class, keyValFilter);
+        }
+        catch (IgniteCheckedException | RuntimeException e) {
+            closeFilter(keyValFilter);
+
+            throw e;
+        }
+    }
+
+    /** */
+    private boolean filter(K k, V v) {
+        try {
+            return scanFilter == null || scanFilter.apply(k, v);
+        }
+        catch (Throwable e) {
+            throw new IgniteException(e);
+        }
+    }
+
+    /** */
+    static void closeFilter(IgniteBiPredicate<?, ?> filter) {
+        if (filter instanceof PlatformCacheEntryFilter)
```

Comment:
Looks like after sendboxing `PlatformCacheEntryFilter` is not `instanceof PlatformCacheEntryFilter` anymore

Reply:
Good catch. Thanks. Fixed.